### ISSUE TITLE
Extending EventEmitter stops module class from checking 

### ISSUE THREAD

**Issue by runn1ng** [facebook/flow#2102]
I have this in flow declaration file for an [external module](https://github.com/node-hid/node-hid)

```javascript
import {EventEmitter} from 'Events';

declare module 'node-hid' {
  
  declare type HIDDeviceDescription = {
    vendorId: number;
    productId: number;
    path: string;
  }

  declare function devices(): Array<HIDDeviceDescription>;
  
  declare class HID extends EventEmitter {
    constructor(path: string): void;
    write(buffer: Array<number>): void;
    pause(): void;
    resume(): void;
    close(): void;
  } 
}
```

This is in my code

```javascript
var device = new HID.HID("123");
device.foobar();
```

`device.foobar` should throw an error, but it doesn't. Removing `extends EventEmitter` fixes that, but then, I cannot use EventEmitter's functions.

**Comment by runn1ng** [comment]
hm, it seems the definitions in flow-typed are using `extends events$EventEmitter`. I don't entirely understand why, but when it works for others, it should work for me :)

And yes, it does work. You can close this issue since it's probably a duplicate of another existing issue.

**Comment by avikchaudhuri** [comment]
@runn1ng Are you saying that the above code typechecks differently when you use `EventEmitter` after import vs. the global (but intended to be private to the lib file) `events$EventEmitter`? I don't think we support importing from one lib file to define another yet. So it's kind of expected.

But this is also kind of a missing feature...

**Comment by runn1ng** [comment]
Yep, that's what I am saying. And with EventEmitter after import, it checks
incorrectly.

So yeah it's probably just a missing feature, but then it should just throw
error and not silently accept.
